django/core/serializers/xml_serializer.py

```diff
@@ -108,14 +111,10 @@ def handle_fk_field(self, obj, field):
         self._start_relational_field(field)
         related_att = getattr(obj, field.attname)
         if related_att is not None:
-            if self.use_natural_foreign_keys and hasattr(
-                field.remote_field.model, "natural_key"
-            ):
-                related = getattr(obj, field.name)
-                # If related object has a natural key, use it
-                related = related.natural_key()
+            natural_key_value = self._resolve_fk_natural_key(obj, field)
+            if natural_key_value:
```

Comment:
keys can be falsey, e.g. 0
```suggestion
            if natural_key_value is not None:
```